lightning-invoice/src/utils.rs

```diff
@@ -50,14 +50,77 @@ use sync::Mutex;
 /// [`ChannelManager::get_phantom_route_hints`]: lightning::ln::channelmanager::ChannelManager::get_phantom_route_hints
 /// [`PhantomRouteHints::channels`]: lightning::ln::channelmanager::PhantomRouteHints::channels
 pub fn create_phantom_invoice<Signer: Sign, K: Deref>(
-	amt_msat: Option<u64>, description: String, payment_hash: PaymentHash, payment_secret:
-	PaymentSecret, phantom_route_hints: Vec<PhantomRouteHints>, keys_manager: K, network: Currency
+	amt_msat: Option<u64>, description: String, payment_hash: PaymentHash, payment_secret: PaymentSecret,
+	phantom_route_hints: Vec<PhantomRouteHints>, keys_manager: K, network: Currency,
 ) -> Result<Invoice, SignOrCreationError<()>> where K::Target: KeysInterface {
+	let description = Description::new(description).map_err(SignOrCreationError::CreationError)?;
+	let description = InvoiceDescription::Direct(&description,);
+	_create_phantom_invoice::<Signer, K>(
+		amt_msat, description, payment_hash, payment_secret, phantom_route_hints, keys_manager, network,
+	)
+}
+
+#[cfg(feature = "std")]
+/// Utility to create an invoice that can be paid to one of multiple nodes, or a "phantom invoice."
```

Comment:
Perhaps better to reference the other docs rather than repeating them here. But could be left to a follow-up.